pkg/controller/perconaservermongodbrestore/perconaservermongodbrestore_controller.go

```diff
@@ -218,24 +218,46 @@ func (r *ReconcilePerconaServerMongoDBRestore) reconcileRestore(cr *psmdbv1.Perc
 	case pbm.StatusError:
 		status.State = psmdbv1.RestoreStateError
 		status.Error = meta.Error
+		if err = reEnablePITR(pbmc, stg, cluster.Spec.Backup.PITR); err != nil {
+			return
+		}
 	case pbm.StatusDone:
 		status.State = psmdbv1.RestoreStateReady
 		status.CompletedAt = &metav1.Time{
 			Time: time.Unix(meta.LastTransitionTS, 0),
 		}
+		if err = reEnablePITR(pbmc, stg, cluster.Spec.Backup.PITR); err != nil {
+			return
+		}
 	case pbm.StatusStarting, pbm.StatusRunning:
 		status.State = psmdbv1.RestoreStateRunning
 	}
 
 	return nil
 }
 
-func runRestore(backup string, storage psmdbv1.BackupStorageSpec, pbmc *backup.PBM) (string, error) {
-	err := pbmc.SetConfig(storage)
+func reEnablePITR(pbm *backup.PBM, stg psmdbv1.BackupStorageSpec, pitr psmdbv1.PITRSpec) (err error) {
+	if !pitr.Enabled {
+		return
+	}
+
+	if err = pbm.SetConfig(stg, pitr); err != nil {
+		return errors.Errorf("failed to re-enable PITR: %w", err)
+	}
+
+	return
+}
+
+func runRestore(backup string, storage psmdbv1.BackupStorageSpec, pbmc *backup.PBM, pitr psmdbv1.PITRSpec) (string, error) {
+	pitr.Enabled = false
+	err := pbmc.SetConfig(storage, pitr)
 	if err != nil {
 		return "", errors.Wrap(err, "set pbm config")
 	}
 
+	// pbm agent doesn't disable pitr immediately so operator must wait for it
+	time.Sleep(2 * time.Minute)
```

Comment:
we can't use `sleep` =)
Please, check required statements on reconcile loops, because operator can be restarted/killed and etc

Reply:
I'm new to the project, so thanks, I'll check this :)

The thing is PBM Agent updates pitr state and disables it with some delay. And sending restore command even with disabled pitr in pbmConfig could fail. (due to the asynchronous pitr updates and commands execution)

I'm not sure if it can be synced without modifications to the PBM Agent.

What do you think?